code/datums/components/seethrough.dm

```diff
@@ -0,0 +1,110 @@
+///A component that lets you turn an object invisible when you're standing on certain relative turfs to it, like behind a tree
+/datum/component/seethrough
+	///List of lists that represent relative coordinates to the source atom
+	var/list/relative_turf_coords
+	///A list of turfs on which we make ourself transparent
+	var/list/watched_turfs
+	///Associate list, with client = trickery_image. Track which client is being tricked with which image
+	var/list/tricked_clients = list()
+
+	///Which alpha do we animate towards?
+	var/target_alpha
+	///How long our fase in/out takes
+	var/animation_time
+
+///Pass a list with lists of coordinates. There's a few templates in seethrough.dm.
+/datum/component/seethrough/Initialize(list/relative_turf_coords, target_alpha = 100, animation_time = 0.5 SECONDS)
+	. = ..()
+
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.relative_turf_coords = relative_turf_coords
+	src.target_alpha = target_alpha
+	src.animation_time = animation_time
+
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/DismantlePerimeter)
```

Comment:
why do you need to do this exactly? couldn't you dismantle and then remantle the perimter?

Reply:
I agree with your timer idea

Reply:
10 seconds? that's quite a while

Reply:
I've lowered it to two seconds. Most forcemovers like meteorslugs take less than a second but I'll doubt anyone will even notice